src/tokenizer.ts

```diff
@@ -1494,37 +1494,78 @@ export class Tokenizer extends DiagnosticEmitter {
   }
 
   readDecimalFloat(): f64 {
-    // TODO: numeric separators (parseFloat can't handle these)
     var text = this.source.text;
-    var pos = this.pos;
```

Comment:
This local variable `pos` was added in #1376 , but made using a helper function awkward. I could either:
- leave things how I have them now
- use a local pos and do `this.pos = pos` before each call to the helper and `pos = this.pos` after each call to the helper
- eliminate the helper function and just inline it all in this function.
- pass a `start` argument into the helper (but that still leaves the problem of returning the end position, since I am already using the return value for indicating whether a separator is present). 

Thoughts?

Reply:
Yeah, that's caching `this.pos` is made some thing little complicated but it's speedup self-bootstrap version of AS. I suggest leave all as is because that cacheing most significant only for `unsafeNext`. Rest methods might leave as is for simplicity